producer_client/spec/lib/animals/resources/fake_cat_spec.rb

```diff
@@ -4,10 +4,40 @@
 
 RSpec.describe Animals::Resources::FakeCat do
   describe '#list' do
-    garfield = OpenStruct.new(name: 'Garfield', age: 39)
-    felix = OpenStruct.new(name: 'Felix', age: 99)
-    sylvester = OpenStruct.new(name: 'Sylvester', age: 76)
+    garfield = OpenStruct.new(id: '1', name: 'Garfield', age: 39)
+    felix = OpenStruct.new(id: '2', name: 'Felix', age: 99)
+    sylvester = OpenStruct.new(id: '3', name: 'Sylvester', age: 76)
+
+    before do
+      fake = described_class.new(nil)
+      fake.create(garfield.to_h)
+      fake.create(felix.to_h)
+      fake.create(sylvester.to_h)
+    end
 
     it_behaves_like 'listing cats', [garfield, felix, sylvester]
   end
+
+  describe '#create' do
+    name = 'Garfield'
+    age = 39
+
+    it_behaves_like 'creating a cat', name, age
+
+    it 'keeps created cats in memory' do
```

Comment:
Beyond just the "creating a cat" behavior, the fake has additional important side effects!